File: 1631-number-of-sub-arrays-with-odd-sum/number-of-sub-arrays-with-odd-sum.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var numOfSubarrays = function(arr) {

    let sum = 0;
    let evenCount = 1;
    let oddCount = 0;
    let odd = 0;
    let mod = Math.pow(10, 9) + 7;

    for(const ele of arr) {

        sum += ele;

        if(sum % 2 == 0) {
            evenCount++;
            odd = ( odd +  oddCount ) % mod;
        }
        else {
            oddCount++;
            odd = (odd + evenCount) % mod;
        }

        
    }

    return odd;
    
};

File: 1631-number-of-sub-arrays-with-odd-sum/number-of-sub-arrays-with-odd-sum.cpp
class Solution {
public:
    int numOfSubarrays(vector<int>& arr) {

        int evenSum = 1;
        int oddSum = 0, count = 0, sum = 0;
        
        int mod = 10e8 + 7;

        for(auto &num : arr) {

            sum += num;

            if(sum % 2) {

                oddSum++;
                count = (count + evenSum) % mod;
            }
            else {
                evenSum++;
                count = (count + oddSum) % mod;
            }
        }

        return count;
    }
};